numpy/ma/core.py

```diff
@@ -3183,17 +3186,19 @@ def __setitem__(self, indx, value):
         locations.
 
         """
-        # 2016.01.15 -- v1.11.0
-        warnings.warn(
-           "Currently, slicing will try to return a view of the data," +
-           " but will return a copy of the mask. In the future, it will try" +
-           " to return both as views. This means that using `__setitem__`" +
-           " will propagate values back through all masks that are present.",
-           FutureWarning
-        )
-
         if self is masked:
             raise MaskError('Cannot alter the masked element.')
+
+        if (isinstance(self.mask, np.ndarray) and
```

Comment:
issuing a FutureWarning every time someone slices a masked array is pretty extreme yeah :-/. Enabling the warn-on-write code here would take a little work, but it shouldn't be *too* bad. And we can allocate a new flag for the different error messages if we want, I think -- the special flag values here are only temporary and not part of the public API, so as long as we have free bits there's no problem using them. (So there'd WARN_ON_WRITE_MA, WARN_ON_WRITE_DIAGONAL, etc. as different flags, and the one place that checks for the WARN_ON_WRITE flag would be extended to check for all of them separately.)